client/src/app/modals/deploy-diagram/DeployDiagramModal.js

```diff
@@ -86,9 +105,38 @@ class DeployDiagramModal extends PureComponent {
     this.updateMenu(isFocusedOnInput);
   }
 
-  validateEndpointUrl = url => {
+  asyncValidateEndpointUrl = async ({ endpointUrl, ...values }) => {
+    const baseUrl = this.getBaseUrl(endpointUrl);
+
+    const payload = {
+      url: `${baseUrl}${DEPLOY_CHECK_SUFFIX}`,
+      auth: this.getAuth(values)
+    };
+
+    let connectionError = await this.props.onAction('deploy-check', payload);
```

Comment:
Why does this return the actual error? I'd assume it returns a simple check:

* { ok: true | false, statusCode: ..., statusText }

If it was to return the error, I'd call it `getEndpointReachableError`.

Reply:
Let it be `{ ok: boolean, error: object }`